### PR TITLE
add `show ip bgp summary` command

### PR DESCRIPTION
**Pull request by yzguy**
Here we parse the JSON into structs for BGP neighbor summary information.

### CODE REVIEW

File: module/bgp.go [hunk 1]
```diff
@@ -727,3 +727,42 @@ func (b *BgpNeighborsEntity) SetDescriptionDefault(name string) bool {
...
+	OutMsgQueue         int     `json:"outMsgQueue"`
+	UnderMaintenance    bool    `json:"underMaintenance"`
+	ASN                 int64   `json:"asn"`
+}
+
+func (b *ShowIPBGPSummary) GetCmd() string {
+	return "show ip bgp summary"
+}
+
+func (s *ShowEntity) ShowIPBGPSummary() ShowIPBGPSummary {
```
Comment (cheynearista) [inline module/bgp.go]: Wondering if returning (ShowIPBGPSummary, error) would be better. Otherwise the caller has no knowledge of an underlying error.

**Comment by yzguy** [comment]
@cheynearista I'm having a tough time wrapping my head around how to create a test to cover the code sending back errors. I'm wondering if the line with the `.Call()` should be the only one sending back an error.

Do you have any suggestions on how to handle a test that triggers those function calls to return errors?

**Comment by cheynearista** [comment]
The DummyConnection you created in lldp_test.go could be leveraged for this....after all it's merely mocking an EapiConnection. I think using the command as the base name for the fixtures is good (i.e. show_lldp_neighbors.json).  Returning error from `Execute()` here would be simple...you could just pre-init `DummyConnection` when you create your mock connection:
```
        dummyConnection := &DummyConnection{err: errors.New("Some Error")}
        dummyNode := &goeapi.Node{}
        dummyNode.SetConnection(dummyConnection)
        show := Show(dummyNode)
        showLldpNeighbors := show.ShowLLDPNeighbors()
```
This would cause `Execute()` to return error....and bubble up to `Call()`. To go even further, you could also (in someway) pull in an invalid json fixture....not exactly sure the best way to do this, but one way is just creating an errorFixture string within DummyConnection and if that's set to anything, read that fixture in.

`ShowLLDPNeighbors()` would then need to return `(ShowLLDPNeighbors, errror)`....which I think makes sense...not sure why I didn't do this originally :-/

Does that make sense?

**Comment by yzguy** [comment]
Yes that makes sense, I'm going to try and take a look at it today. Thanks!

**Comment by yzguy** [comment]
@cheynearista I finally got around to this again, put a quick test in with your feedback (easier than I thought). Code coverage came back with a 100%

**Comment by cheynearista** [comment]
@yzguy Merged. Thanks!

**Pull request merged**